examples/apps/ai-collab/src/components/TaskCard.tsx

```diff
@@ -195,30 +185,93 @@ export function TaskCard(props: {
 											autoHideDuration: 5000,
 										});
 
-										const response = await editTask(task, query);
+										// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+										const currentBranch = TreeAlpha.branch(props.sharedTreeBranch.root)!;
+										const aiCollabBranch = currentBranch.fork() as TreeViewAlpha<
+											typeof SharedTreeAppState
+										>;
+										const parentTaskGroup = Tree.parent(
+											// eslint-disable-next-line @typescript-eslint/no-non-null-assertion, -- Note that two levels up from the task is the task group node.
+											Tree.parent(props.sharedTreeTask)!,
+										) as SharedTreeTaskGroup;
 
-										setIsAiTaskRunning(false);
+										// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+										const aiCollabBranchTask = aiCollabBranch.root.taskGroups
+											.find((taskGroup) => taskGroup.id === parentTaskGroup.id)!
+											.tasks.find((task) => task.id === props.sharedTreeTask.id)!;
+
+										console.log("aiCollabBranchTask:", { ...aiCollabBranchTask });
 
-										if (response.success) {
+										let response: AiCollabSuccessResponse | AiCollabErrorResponse;
+										try {
+											response = await aiCollab({
+												openAI: {
+													client: new OpenAI({
+														apiKey: process.env.NEXT_PUBLIC_OPENAI_API_KEY,
+														dangerouslyAllowBrowser: true,
```

Comment:
Not immediately evident why this is defaulted to true. Maybe add a comment explaining why